flexmeasures/api/common/tests/test_schemas.py

```diff
@@ -38,12 +45,16 @@ def test_duration_field_nominal_grounded(
     """Nominal durations are tricky:
     https://en.wikipedia.org/wiki/Talk:ISO_8601/Archive_2#Definition_of_Duration_is_incorrect
     We want to test if we can ground them as expected.
-    We use a particular datetime to ground, in a leap year february.
+    We use a particular datetime to ground, in a leap year February.
+    For the Europe/Amsterdam timezone, daylight saving time started on March 29th 2020.
+    # todo: the commented out tests would work if isodate.parse_duration would have the option to stop coercing ISO 8601 days into datetime.timedelta days
```

Comment:
If I find time, I may open up a ticket and PR for the isodate package, because their parsing of P1D to a datetime.timedelta(days=1) does not reflect the ISO 8601 standard. A simple fix would be to have `isodate.parse_duration` accept a boolean parameter that forces the return type to `isodate.Duration`. From that object, the `pandas.DateOffset` can be constructed as I've done in `schemas.py`. As far as I have seen, the pandas DateOffset is the only implementation that can correctly apply the logic as instructed by ISO 8601 (even the `dateutil.relativedelta.relativedelta` on which the `pandas.DateOffset` is based doesn't account for the transition to daylight saving time).

Reply:
Thanks for the extra content here.